examples/Nat/TcpState.cs

```diff
@@ -105,12 +105,15 @@ private void TransitionState(ref TcpDirectionalState state, bool syn, bool fin)
 
     private void TransitionState(ref TcpDirectionalState state, bool ack)
     {
-      if (state == TcpDirectionalState.Syn)
-        state = TcpDirectionalState.SynAck;
-      else if (state == TcpDirectionalState.Fin)
-      {
-        CloseTime = DateTime.Now;
-        state = TcpDirectionalState.FinAck;
+      if (ack)

```

Comment:
I think I lost the flow of this function and how it fits in the big picture -- why must `ack==true` for the function to have any effect? Perhaps add a comment to the function to sketch its role wrt ack? Thanks